controllers/rollingupgrade_controller.go

```diff
@@ -704,9 +705,43 @@ func (r *RollingUpgradeReconciler) Process(ctx *context.Context,
 		return
 	}
 
+	//Validation step: check if all the nodes have the latest launchconfig.
+	r.info(ruObj, "Validating the launch config of nodes and ASG")
+	if err := r.validateNodesLaunchConfig(ruObj); err != nil {
+		r.error(ruObj, err, "Launch config validation failed")
+		r.finishExecution(StatusError, nodesProcessed, ctx, ruObj)
+		return
+	}
+	r.info(ruObj, "Launch config validation succeeded")
+
 	r.finishExecution(StatusComplete, nodesProcessed, ctx, ruObj)
 }
 
+//Check if ec2Instances and the ASG have same launch config.
+func (r *RollingUpgradeReconciler) validateNodesLaunchConfig(ruObj *upgrademgrv1alpha1.RollingUpgrade) error {
+	//Get ASG launch config
+	err := r.populateAsg(ruObj)
+	if err != nil {
+		return errors.New("Unable to populate the ASG object")
+	}
+	asg, err := r.GetAutoScalingGroup(ruObj.Name)
+	if err != nil {
+		return fmt.Errorf("Unable to load ASG with name: %s", ruObj.Name)
+	}
+	launchDefinition := NewLaunchDefinition(asg)
+	launchConfigASG := *(launchDefinition.launchConfigurationName)
+
+	//Get ec2 instances and their launch configs.
+	ec2instances := asg.Instances
+	for _, ec2Instance := range ec2instances {
+		if launchConfigASG != aws.StringValue(ec2Instance.LaunchConfigurationName) {
```

Comment:
Can you check of the `ec2Instance` object has a `LaunchTemplateID` in it?

Basically, if the launchDefinition has a `launchConfig`, the above check is correct. If the launchDefinition has a `launchTemplate`, the check should be modified for that.

Reply:
ec2Instance doesn't have Launch template when accessed through **asg.Instances**
`"Instance": "{AvailabilityZone: "xx-yy", HealthStatus: "Healthy", InstanceId: "i-abc", InstanceType: "aa.bb",  LaunchConfigurationName: "nodes-xxyy",  LifecycleState: "InService",  ProtectedFromScaleIn: false}"`

However, when I access the ec2Instance using **NodeSelector()**, I see both launch config and launch template fields.
`"Instance": [{"AvailabilityZone":"us-west-2c","HealthStatus":"Healthy","InstanceId":"i-xxx","InstanceType":"xx.yy","LaunchConfigurationName":"sbadiger-instance-manager-nodes-xxx","LaunchTemplate":null,"LifecycleState":"InService","ProtectedFromScaleIn":false,"WeightedCapacity":null}]`

Reply:
Addressed this in the latest commit. LaunchTemplateID is also validated.